### PR TITLE
Uploading files for new integration tests

### PR DESCRIPTION
**Pull request by kylacochrane**
This PR adds additional integration tests and files pertaining to:

1. Filtering the query sample cluster addresses for the nf-iridanext plugin
2. Ensuring query samples are removed from the pipeline if meta.id ≠ JSON Key 
3. Ensuring pipeline is halted if reference sample meta.id ≠ JSON Key

### CODE REVIEW

File: tests/pipelines/main.nf.test [hunk 1]
```diff
@@ -33,8 +32,80 @@ nextflow_pipeline {
...
             assert actual_calls.text == expected_calls.text
+
+            // Check IRIDA Next JSON output
+            assert path("$launchDir/results/iridanext.output.json").json == path("$baseDir/tests/data/irida/test_iridanext.output.json").json
+
+            def iridanext_json = path("$launchDir/results/iridanext.output.json").json
+            def iridanext_samples = iridanext_json.files.samples
+            def iridanext_metadata = iridanext_json.metadata.samples
+
+            assert iridanext_metadata.sampleQ."address" == "1.1.3"
```
Comment (apetkau) [inline tests/pipelines/main.nf.test:43]: Could you add an assert in this case to make sure that `sampleQ` is the only sample listed in the metadata?

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks great Kyla. Thanks so much. Just one comment 😄

File: tests/pipelines/main.nf.test (hunk 1 above)
Comment (kylacochrane) [reply tests/pipelines/main.nf.test:44]: Sure 😄 : [7feaee8](https://github.com/phac-nml/gasnomenclature/pull/10/commits/7feaee8f6c38a61ca6ef8d4916b808aff9f4645e)

**Review by apetkau — APPROVED** [review]
Looks great. Thanks so much 😄

**Pull request merged**